### PR TITLE
vam-513-orqqpl-init-p

### PR DESCRIPTION
**Pull request by pcchanghawaii**
hard code result MST & head/neck cancer as empty '' for now

### CODE REVIEW

File: emulation/models/orqqpl-init-pt.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        const returnArray = [];
+        if (res) {
+            const sc = _.get(res, 'serviceConnected[0].isServiceConnected') ? 1 : 0;
+            const facilityUrn = _.get(res, 'preferredFacility.id', '');
+            const aoExpo = _.get(res, 'treatmentFactors[0].agentOrangeExposure[0].exposureIndicated', '').toUpperCase()
+                === 'YES' ? 1 : 0;
+            const radiExpo = _.get(res, 'treatmentFactors[0].radiationExposure[0].exposureIndicated', '').toUpperCase()
+                === 'YES' ? 1 : 0;
+            const gulfWar = _.get(res, 'persianGulfService[0].serviceIndicated', '').toUpperCase()
+                === 'YES' ? '1' : '';
```
Comment (mfuroyama) [inline emulation/models/orqqpl-init-pt.js]: Looks like the results here are either '1' or '' where the other results are either '1' or '0'. Just checking if this was intentional or not.

File: emulation/models/orqqpl-init-pt.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            const cv = _.get(res, 'combatService[0].serviceIndicated') ? 1 : 0;
+            const shad = _.get(res, 'proj112Shad') ? 1 : 0;
+
+            let bid = _.get(res, 'socialSecurityNumber', '');
+            if (bid.length === 9) {
+                bid = bid.substring(5);
+            }
+            returnArray.push(serviceUtils.getIdFromUrn(facilityUrn));
+            returnArray.push(_.get(res, 'dateOfDeath', ''));
```
Comment (mfuroyama) [inline emulation/models/orqqpl-init-pt.js]: Oh, one thing to note that I forgot to note in the other pull requests: the fields listed as 'DATE' in the MongoDB schemas will come back in the object as JS Date objects. Stuff to keep in mind, especially when determining the correct means to format output values.

File: emulation/models/orqqpl-init-pt.js (hunk 1 above)
Comment (pcchanghawaii) [reply emulation/models/orqqpl-init-pt.js]: yes. this was intended. 
The following are the result for the non-emulated RPC call.
25: [ 2957, '', 0, 0, 1, '', '0113', '', '', 0, 0 ]
        expect(res.result[0]).toEqual(2957); // VISTA HEALTH CARE 4-2957
        expect(res.result[1]).toEqual(''); // death indicator
        expect(res.result[2]).toEqual(0); // service connected
        expect(res.result[3]).toEqual(0); // agent orange exposure
        expect(res.result[4]).toEqual(1); // ionizing radiation exposure
        expect(res.result[5]).toEqual(''); // gulf war exposure
        expect(res.result[6]).toEqual('0113'); // last4 ssn, need this to reconstitute GMPDFN on return
        expect(res.result[7]).toEqual(''); // head/neck cancer
        expect(res.result[8]).toEqual(''); // MST, MST HISTORY
        expect(res.result[9]).toEqual(0); // CV, combat service indicated?
        expect(res.result[10]).toEqual(0); // SHAD, proj 112/shad

11: [ 2957, '', 0, 0, 0, '', 5432, '', '', 0, 0 ]
27: [ 2957, '', 0, 0, 0, '', '0029', '', '', 0, 0 ]

**Comment by pcchanghawaii** [comment]
just realized this pull request was omitted earlier. I have updated it for the date format. I'm not quite sure the death indicator format though (date or 1/0).  documented in the code.

File: emulation/models/orqqpl-init-pt.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    }],
+    labels: ['META'],
+    services: ['patientMetaService.getInitParameters'],
+    parameters: [],
+    invokeRPC: (args, session, callback) => {
+        const urnObj = {
+            fileType: 2,
+            id: args[0],
+            isNational: false,
+            station: '999',
```
Comment (mfuroyama) [inline emulation/models/orqqpl-init-pt.js]: Let's remove the hard-coding and use the `session.stationId` to set this

Comment (mfuroyama) [reply emulation/models/orqqpl-init-pt.js]: You know what? I'll fix this...it's simple

**Pull request merged**